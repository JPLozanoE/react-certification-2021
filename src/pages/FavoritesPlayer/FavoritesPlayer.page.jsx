import React from 'react';
import { VideoPlayerView } from '../../components/Home/VideoPlayerView';

export const FavoritesPlayerPage = () => <VideoPlayerView favorites />;
