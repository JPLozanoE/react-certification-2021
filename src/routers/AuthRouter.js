/* eslint-disable import/extensions */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { Redirect, Route, Switch } from 'react-router';
import { LoginScreen } from '../components/Auth/LoginScreen';
import { RegisterScreen } from '../components/Auth/RegisterScreen';
import { AuthPage } from '../pages/Auth/Auth.page';

export const AuthRouter = () => {
  return (
    <AuthPage>
      <div className="auth__box-container animate__animated animate__fadeIn animate__faster">
        <Switch>
          <Route path="/auth/login" component={LoginScreen} />
          <Route path="/auth/register" component={RegisterScreen} />
          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </AuthPage>
  );
};
