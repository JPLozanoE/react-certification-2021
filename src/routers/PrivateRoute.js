/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { Redirect, Route } from 'react-router';

export const PrivateRoute = ({ isAuth, component: Component, ...rest }) => {
  // localStorage.setItem('lastPath', rest.location.pathname);
  return (
    <Route
      {...rest}
      component={(props) =>
        isAuth ? <Component {...props} /> : <Redirect to="/auth/login" />
      }
    />
  );
};
