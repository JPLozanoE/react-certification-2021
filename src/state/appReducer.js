import { types } from '../types/types';

export const appReducer = (state = {}, action) => {
  switch (action?.type) {
    case types.changeSearchQuery:
      return {
        ...state,
        search: action.payload,
      };

    case types.setSelectedVideo:
      return {
        ...state,
        selectedVideo: action.payload,
      };

    case types.toggleTheme:
      return {
        ...state,
        isDarkTheme: !state.isDarkTheme,
      };

    case types.setRecommendedVideos:
      return {
        ...state,
        videos: action.payload,
      };

    case types.login:
      return {
        ...state,
        auth: {
          isLogged: true,
          displayName: action.payload.displayName,
          uid: action.payload.uid,
        },
      };

    case types.logout:
      return {
        ...state,
        auth: {
          isLogged: false,
          displayName: '',
          uid: null,
        },
      };

    case types.addFavoriteVideo:
      return {
        ...state,
        favoriteVideos: [
          ...state.favoriteVideos,
          { ...action.payload, isFavorite: true },
        ],
      };

    case types.deleteFavoriteVideo:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: false } : video
        ),
        favoriteVideos: state.favoriteVideos.filter(
          (favoriteVideo) => favoriteVideo.id.videoId !== action.payload
        ),
      };

    case types.updateRecommendedVideos:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: true } : video
        ),
      };

    default:
      return state;
  }
};
