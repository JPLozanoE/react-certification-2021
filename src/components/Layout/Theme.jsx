import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { AppContext } from '../../state/AppContext';

const Theme = ({ children }) => {
  const {
    state: { isDarkTheme },
  } = useContext(AppContext);
  const theme = {
    main: {
      ...(isDarkTheme ? { backgroundColor: '#181818' } : { backgroundColor: '#ffffff' }),
    },
    titles: {
      ...(isDarkTheme ? { color: '#ffffff' } : { color: '#00000' }),
    },
    subtitles: {
      ...(isDarkTheme ? { color: 'rgb(170,170,170)' } : { color: 'rgb(96,96,96)' }),
    },
  };

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default Theme;
