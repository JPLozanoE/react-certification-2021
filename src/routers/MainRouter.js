/* eslint-disable react/jsx-filename-extension */
import React, { useContext, useEffect, useState } from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import GlobalStyles, { FullWidthContainer } from '../__globalStyles';
import { AppContext } from '../state/AppContext';
import { searchYouTube } from '../utils/searchYoutube';
import Layout from '../components/Layout/Layout.component';
import HomePage from '../pages/Home/Home.page';
import { Navbar } from '../components/Home/Navbar';
import { types } from '../types/types';
import { VideoPage } from '../pages/Video/Video.page';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { AuthRouter } from './AuthRouter';
// import { Favorites } from '../components/Favorites/Favorites';
import { FavoritesPage } from '../pages/Favorites/Favorites.page';
import { FavoritesPlayerPage } from '../pages/FavoritesPlayer/FavoritesPlayer.page';

const RedirectImprov = () => <Redirect to="/" />;

function MainRouter() {
  const [error, setError] = useState(false);
  const {
    state: {
      selectedVideo,
      search,
      auth: { isLogged },
    },
    dispatch,
  } = useContext(AppContext);

  console.log(isLogged);

  useEffect(() => {
    const fetchVideos = async () => {
      const res = await searchYouTube(search);
      if (res.error) setError(true);
      dispatch({ type: types.setRecommendedVideos, payload: res.items });
    };
    fetchVideos();
  }, [search, dispatch]);

  return (
    <BrowserRouter>
      <GlobalStyles />
      <Layout>
        <Navbar />
        <FullWidthContainer>
          <Switch>
            <PublicRoute path="/auth" component={AuthRouter} isAuth={isLogged} />
            {error ? (
              <p>Imposible mostrar en este momento</p>
            ) : (
              <>
                <Route exact path="/">
                  <HomePage />
                </Route>
                <Route exact path="/video/:videoId">
                  {selectedVideo !== null ? <VideoPage /> : <Redirect to="/" />}
                </Route>
                <PrivateRoute
                  exact
                  path="/favorites"
                  component={FavoritesPage}
                  isAuth={isLogged}
                />
                <PrivateRoute
                  exact
                  path="/video/favorites/:videoId"
                  component={
                    selectedVideo !== null ? FavoritesPlayerPage : RedirectImprov
                  }
                  isAuth={isLogged}
                />
              </>
            )}
          </Switch>
        </FullWidthContainer>
      </Layout>
    </BrowserRouter>
  );
}

export default MainRouter;
