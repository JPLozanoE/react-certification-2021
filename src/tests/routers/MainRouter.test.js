/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Theme from '../../components/Layout/Theme';
import { demoState, singleVideo } from '../fixtures/demoState';
import { AppContext } from '../../state/AppContext';
import MainRouter, { RedirectImprov } from '../../routers/MainRouter';
import { PrivateRoute } from '../../routers/PrivateRoute';
import { FavoritesPlayerPage } from '../../pages/FavoritesPlayer/FavoritesPlayer.page';

describe('Main Router', () => {
  let wrapper;

  test('Should render properly when in /auth/register', () => {
    const providerValues = {
      state: { ...demoState, selectedVideo: singleVideo },
      dispatch: jest.fn(),
    };
    const history = createMemoryHistory();
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router history={history}>
            <MainRouter />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.getByText(demoState.videos[0].snippet.channelTitle)
    ).toBeInTheDocument();
  });

  test('Should render properly ', () => {
    const providerValues = {
      state: { ...demoState },
      dispatch: jest.fn(),
    };
    const history = createMemoryHistory();
    history.push('/video/favorites/5');
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router history={history}>
            <PrivateRoute
              exact
              path="/video/favorites/:videoId"
              component={
                providerValues.state.selectedVideo !== null
                  ? FavoritesPlayerPage
                  : RedirectImprov
              }
              isAuth
            />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(
      wrapper.queryByText(demoState.favoriteVideos[0].snippet.title)
    ).not.toBeInTheDocument();
  });
});
