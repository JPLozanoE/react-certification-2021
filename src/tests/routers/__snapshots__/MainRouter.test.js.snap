// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Router Should render properly when in /auth/register 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-hKFxyN fOcJjF"
      >
        <div
          class="sc-kEqXSa fRhAbd topnav"
          id="myTopnav"
        >
          <a
            class="sc-iqAclL jRYrJJ"
            href="/"
          >
            Home
          </a>
          <div
            class="sc-fujyAs iyNEPm topnav"
            id="myTopnav"
          >
            <input
              class="sc-pNWdM cLsCj"
              placeholder="Enter your search..."
              value=""
            />
            <button
              class="sc-jrsJWt foBoLi"
              name="search"
            >
              Search
            </button>
          </div>
          <div
            class="sc-hBMUJo ldJdCL"
          >
            <p
              class="sc-crzoAE kbehzr"
              href="#about"
            >
              Dark mode
            </p>
            <div
              class="sc-fnVZcZ cvvNjz"
            >
              <div
                class="sc-dIsUp bctaC"
              >
                <input
                  checked=""
                  class="sc-ksluID jYamWL"
                  id="checkbox"
                  type="checkbox"
                />
                <label
                  class="sc-bqGGPW jLTksx"
                  for="checkbox"
                />
              </div>
              <a
                class="sc-iqAclL jRYrJJ"
                href="/auth"
              >
                Login
              </a>
            </div>
          </div>
        </div>
        <div
          class="sc-dlnjwi eueSIH"
        >
          <div
            class="sc-eCApnc coObtW"
          >
            <div
              class="sc-jSFjdj jLPDRD"
            >
              <img
                alt="img"
                aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
                height="150px"
                src="https://i.ytimg.com/vi/yFq-c9ZpmRs/hqdefault.jpg"
                style="object-fit: cover;"
                width="100%"
              />
              <div
                style="padding: 10px;"
              >
                <div>
                  <h3
                    class="sc-gKAaRy cAjJAT"
                  >
                    Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
                  </h3>
                </div>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  MLB
                </p>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  Publicado el 
                  2021-08-21T03:02:50Z
                </p>
              </div>
            </div>
            <div
              class="sc-jSFjdj jLPDRD"
            >
              <img
                alt="img"
                aria-label="State of Emergency As Hurricane Henri Approaches New York"
                height="150px"
                src="https://i.ytimg.com/vi/EqG9gWbwxlw/hqdefault.jpg"
                style="object-fit: cover;"
                width="100%"
              />
              <div
                style="padding: 10px;"
              >
                <div>
                  <h3
                    class="sc-gKAaRy cAjJAT"
                  >
                    State of Emergency As Hurricane Henri Approaches New York
                  </h3>
                </div>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  NBC News
                </p>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  Publicado el 
                  2021-08-21T23:30:18Z
                </p>
              </div>
            </div>
            <div
              class="sc-jSFjdj jLPDRD"
            >
              <img
                alt="img"
                aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
                height="150px"
                src="https://i.ytimg.com/vi/LqiDJh4zxAc/hqdefault.jpg"
                style="object-fit: cover;"
                width="100%"
              />
              <div
                style="padding: 10px;"
              >
                <div>
                  <h3
                    class="sc-gKAaRy cAjJAT"
                  >
                    MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
                  </h3>
                </div>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  Magic Secrets Revealed
                </p>
                <p
                  class="sc-iCoGMd iGounM"
                >
                  Publicado el 
                  2019-09-05T21:00:03Z
                </p>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-hKFxyN fOcJjF"
    >
      <div
        class="sc-kEqXSa fRhAbd topnav"
        id="myTopnav"
      >
        <a
          class="sc-iqAclL jRYrJJ"
          href="/"
        >
          Home
        </a>
        <div
          class="sc-fujyAs iyNEPm topnav"
          id="myTopnav"
        >
          <input
            class="sc-pNWdM cLsCj"
            placeholder="Enter your search..."
            value=""
          />
          <button
            class="sc-jrsJWt foBoLi"
            name="search"
          >
            Search
          </button>
        </div>
        <div
          class="sc-hBMUJo ldJdCL"
        >
          <p
            class="sc-crzoAE kbehzr"
            href="#about"
          >
            Dark mode
          </p>
          <div
            class="sc-fnVZcZ cvvNjz"
          >
            <div
              class="sc-dIsUp bctaC"
            >
              <input
                checked=""
                class="sc-ksluID jYamWL"
                id="checkbox"
                type="checkbox"
              />
              <label
                class="sc-bqGGPW jLTksx"
                for="checkbox"
              />
            </div>
            <a
              class="sc-iqAclL jRYrJJ"
              href="/auth"
            >
              Login
            </a>
          </div>
        </div>
      </div>
      <div
        class="sc-dlnjwi eueSIH"
      >
        <div
          class="sc-eCApnc coObtW"
        >
          <div
            class="sc-jSFjdj jLPDRD"
          >
            <img
              alt="img"
              aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
              height="150px"
              src="https://i.ytimg.com/vi/yFq-c9ZpmRs/hqdefault.jpg"
              style="object-fit: cover;"
              width="100%"
            />
            <div
              style="padding: 10px;"
            >
              <div>
                <h3
                  class="sc-gKAaRy cAjJAT"
                >
                  Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
                </h3>
              </div>
              <p
                class="sc-iCoGMd iGounM"
              >
                MLB
              </p>
              <p
                class="sc-iCoGMd iGounM"
              >
                Publicado el 
                2021-08-21T03:02:50Z
              </p>
            </div>
          </div>
          <div
            class="sc-jSFjdj jLPDRD"
          >
            <img
              alt="img"
              aria-label="State of Emergency As Hurricane Henri Approaches New York"
              height="150px"
              src="https://i.ytimg.com/vi/EqG9gWbwxlw/hqdefault.jpg"
              style="object-fit: cover;"
              width="100%"
            />
            <div
              style="padding: 10px;"
            >
              <div>
                <h3
                  class="sc-gKAaRy cAjJAT"
                >
                  State of Emergency As Hurricane Henri Approaches New York
                </h3>
              </div>
              <p
                class="sc-iCoGMd iGounM"
              >
                NBC News
              </p>
              <p
                class="sc-iCoGMd iGounM"
              >
                Publicado el 
                2021-08-21T23:30:18Z
              </p>
            </div>
          </div>
          <div
            class="sc-jSFjdj jLPDRD"
          >
            <img
              alt="img"
              aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
              height="150px"
              src="https://i.ytimg.com/vi/LqiDJh4zxAc/hqdefault.jpg"
              style="object-fit: cover;"
              width="100%"
            />
            <div
              style="padding: 10px;"
            >
              <div>
                <h3
                  class="sc-gKAaRy cAjJAT"
                >
                  MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
                </h3>
              </div>
              <p
                class="sc-iCoGMd iGounM"
              >
                Magic Secrets Revealed
              </p>
              <p
                class="sc-iCoGMd iGounM"
              >
                Publicado el 
                2019-09-05T21:00:03Z
              </p>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
